lib/ansible/inventory/manager.py

```diff
@@ -130,7 +130,7 @@ def split_host_pattern(pattern):
                 '''), pattern, re.X
             )
 
-    return [p.strip() for p in patterns]
+    return [p.strip() for p in patterns if p.strip()]
```

Comment:
Not something you can use now, but since I know the rdundancy of calling `strip()` twice probably drove you nuts.  In python-3.8+, you can use the walrus operator:
```
return [stripped_pat for pat in patterns if (stripped_pat := pat.strip())]
```
That particular example might be a bit obfuscated, though.  I'm not sure if it's just the unfamiliarity of the walrus operator that makes it less readable or that it leads to too much going on in too little space.  Time will tell.

Reply:
Yes, I did too many permutations and measurements to avoid the "double stripping" but didn't see any significant performance difference.

I have seen the walrus operator a bit but don't fully get it yet. I have seen a few examples (like this one) that seem hard to fully understand. So I either have more reading to do, or the syntax is too dense/magic.

As you said, time will tell. 😄 Thank you for the suggestion, though.